src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -6422,4 +6789,400 @@ public static class D
+        Assert.Equal(locationSpecifier, model.GetInterceptableLocation(node));
+        Assert.Equal(locationSpecifier.GetHashCode(), model.GetInterceptableLocation(node)!.GetHashCode());
+
+        // If Data changes it might be the case that 'SourceText.GetContentHash()' has changed algorithms.
+        // In this case we need to adjust the SourceMethodSymbolWithAttributes.DecodeInterceptsLocationAttribute impl to remain compatible with v1 and consider introducing a v2 which uses the new content hash algorithm.
+        AssertEx.Equal("jB4qgCy292LkEGCwmD+R6FIAAABQcm9ncmFtLmNz", locationSpecifier.Data);
+        AssertEx.Equal("""[global::System.Runtime.CompilerServices.InterceptsLocationAttribute(1, "jB4qgCy292LkEGCwmD+R6FIAAABQcm9ncmFtLmNz")]""", locationSpecifier.GetInterceptsLocationAttributeSyntax());
+    }
+
+    [Fact]
+    public void Checksum_03()
+    {
+        // Invalid base64
+        var interceptors = CSharpTestSource.Parse($$"""
+            using System;
+            using System.Runtime.CompilerServices;
+
+            static class Interceptors
+            {
+                [InterceptsLocation(1, "jB4qgCy292LkEGCwmD+R6AcAAAAJAAAAUHJvZ3JhbS5jcw===")]
+                public static void M1() => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation([interceptors, CSharpTestSource.Parse(s_attributesSource.text, s_attributesSource.path, RegularWithInterceptors)]);
+        comp.VerifyEmitDiagnostics(
+            // Interceptors.cs(6,6): error CS9230: The data argument to InterceptsLocationAttribute is not in the correct format.
+            //     [InterceptsLocation(1, "jB4qgCy292LkEGCwmD+R6AcAAAAJAAAAUHJvZ3JhbS5jcw===")]
+            Diagnostic(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, "InterceptsLocation").WithLocation(6, 6));
+    }
+
+    [Fact]
+    public void Checksum_04()
+    {
+        // Test invalid UTF-8 encoded to base64
+
+        var builder = new BlobBuilder();
+        // all zeros checksum and zero position
+        builder.WriteBytes(value: 0, byteCount: 20);
+
+        // write invalid utf-8
+        builder.WriteByte(0xc0);
+
+        var base64 = Convert.ToBase64String(builder.ToArray());
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            using System;
+            using System.Runtime.CompilerServices;
+
+            static class Interceptors
+            {
+                [InterceptsLocation(1, "{{base64}}")]
+                public static void M1() => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation([interceptors, CSharpTestSource.Parse(s_attributesSource.text, s_attributesSource.path, RegularWithInterceptors)]);
+        comp.VerifyEmitDiagnostics(
+            // Interceptors.cs(6,6): error CS9233: Cannot intercept a call in file '�' because a matching file was not found in the compilation.
```

Comment:
>� [](http://example.com/codeflow?start=84&length=1)

I'm not sure how much we care, but technically isn't this character a valid filename?
It would be better to do the UTF8 decoding without fallback, so that decoding fails outright rather than silently produce a substituted string.

Reply:
I didn't want to go to the trouble, but the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=net-8.0#system-text-utf8encoding-ctor) convinced me we should go ahead and throw on invalid bytes.